### DISCUSSION TITLE
Velocity attribution in orbit.py (Orbit Trajectory) multirotor code example

### DISCUSSION THREAD

**Discussion by arielingrid** [microsoft/AirSim#3524] [Support Q&A]
In **orbit.py** airsim example code for orbit trajectory, I would like to understand why the calculation of velocity (that will be passed to **moveByVelocityAsync** func) is done as the subtraction of coordinate position values?
![velocity](https://user-images.githubusercontent.com/80911812/111663156-4d5ff580-87ef-11eb-915f-f90bf570453f.png)
![moveVelocity](https://user-images.githubusercontent.com/80911812/111667036-14c21b00-87f3-11eb-9b14-f81963336daa.png)

Position coordinates is returned by the following function:
![position](https://user-images.githubusercontent.com/80911812/111666899-ed6b4e00-87f2-11eb-89a6-48ac1bcf0c6e.png)

  
Ubuntu 18.04, AirSim v1.4.0

**Comment by jonyMarino** [comment]
Hi @arielingrid! I didn't go too deep into the code, but I understand that it is correct because the speed is that distance per second, if the modulus of the distance is always the same, then the modulus of the speed in that trajectory will be too

**Reply by arielingrid** [reply]
Hi @jonyMarino, thanks for reply.
The problem is that the code attributes the difference of distances (lookahead_x, pos.x_val, etc) to a velocity (vx, vy). Velocity and distance have different units (m/s and m). One possible explanation could be the fact that the time would be 1 sec. Still don't understanding...
